docs/design.md

```diff
@@ -82,59 +86,115 @@ library or framework that they instrument. Some examples include:
+Bytecode instrumentations must be implemented following
 the proper attribute annotation so that the native CLR Profiler implementation
 can inject them at runtime. Some examples include:
 
   - [GraphQL](../src/OpenTelemetry.AutoInstrumentation/Instrumentations/GraphQL)
+  - [Logger](../src/OpenTelemetry.AutoInstrumentation/Instrumentations/Logger/)
+  - [MongoDb](../src/OpenTelemetry.AutoInstrumentation/Instrumentations/MongoDb/)
+
+For both kind of instrumentations the instrumentations are enabled only when the
+targeted modules are loaded into the targeted application.
+
+## Architecture
+
+The main components of the project are:
+
+- [**Loader**](../src/OpenTelemetry.AutoInstrumentation.Loader):
+Managed library that bootstraps the **Managed Profiler** code into the
+targeted application extending the
+load paths to include the folders with the OpenTelemetry .NET SDK
+and the instrumentations to be injected into the application.
+
+- [**Managed Profiler**](../src/OpenTelemetry.AutoInstrumentation):
+Contains the code to set up the OpenTelemetry .NET SDK and configured instrumentations,
+as well as support code to run and implement bytecode instrumentations.
+
+- [**CLR Profiler DLL**](../src/OpenTelemetry.AutoInstrumentation.Native):
+Native component that implements a CLR Profiler. The CLR Profiler is used to
+modify the application [intermediate language](https://en.wikipedia.org/wiki/Common_Intermediate_Language)
+(IL), including the IL of packages used by the application, to add and collect
+observability data.
+On the .NET Framework the CLR Profiler DLL is also responsible for injecting
+the **Loader** (see above) during the application start up.
 
 ![Overview](./images/architecture-overview.png)
 
-### Injecting the OpenTelemetry .NET SDK and instrumentations
+### Bootstrapping
+
+While the initial mechanism for bootstrapping the OpenTelemetry .NET SDK differs
+between .NET and .NET Framework applications, giving that the [host startup hook](https://github.com/dotnet/runtime/blob/main/docs/design/features/host-startup-hook.md)
+is not available on the .NET Framework,
+in both cases the initialization is done by creating one instance of the
+`OpenTelemetry.AutoInstrumentation.Loader.Startup` class from the Loader assembly.
+When creating this instance the static constructor of the type performs
+the following actions:
+
+- Adds a handler to the
+[`AssemblyResolve` event](https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyresolve?view=net-5.0),
+so that it can add any assembly needed by the SDK itself or by any instrumentation.
+- Runs, through reflection, the `Initialization` method from
+  the `OpenTelemetry.AutoInstrumentation.Instrumentation` type
+  from the Managed Profiler assembly.
+  - The `Initialization` code bootstraps the OpenTelemetry .NET SDK,
+    adding configured processors, exporters, and so on,
+    and setting the mechanisms to enable any configured source instrumentations.
```

Comment:
```suggestion
1.  Adds a handler to the [`AssemblyResolve` event](https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyresolve?view=net-5.0),
so that it can add any assembly needed by the SDK itself or by any instrumentation.
2. Runs, through reflection, the `Initialization` method from
  the `OpenTelemetry.AutoInstrumentation.Instrumentation` type
  from the Managed Profiler assembly.
  a. The `Initialization` code bootstraps the OpenTelemetry .NET SDK,
    adding configured processors, exporters, and so on,
    and setting the mechanisms to enable any configured source instrumentations.
```